contrib/go-redis/redis/redis_test.go

```diff
@@ -313,3 +313,58 @@ func TestAnalyticsSettings(t *testing.T) {
 		assertRate(t, mt, 0.0, WithAnalyticsRate(0.0))
 	})
 }
+
+type testKey string
+
+func TestWithContextRace(t *testing.T) {
+	opts := &redis.Options{Addr: "127.0.0.1:6379"}
```

Comment:
This makes it seem like it has to be an integration test. Does it? If no, can it work without having Redis running?

Reply:
Yeah but now you have two tests which more or less do the same thing. What you had before was perfect because it was a regression test and is the right thing to do when fixing any bug. It was also short and easy to understand.